tutorials/12-distributed-tracing.md

```diff
@@ -0,0 +1,149 @@
+
+## Table of Contents
+
+1. [Prerequisites](#prerequisites)
+2. [Step 1: Preparing the Jaeger Configuration File](#step-1-preparing-the-jaeger-configuration-file)
+3. [Step 2: Preparing the OpenTelemetry Collector File](#step-2-preparing-the-opentelemetry-collector-file)
+4. [Step 3: Configuring Model and Monitoring](#step-3-configuring-model-and-monitoring)
+
+## Prerequisites
+
+- A Kubernetes environment with GPU support, as set up in the
+  [00-install-kubernetes-env tutorial](00-install-kubernetes-env.md).
+- Helm installed on your system.
+- Access to a HuggingFace token (`HF_TOKEN`).
+- A self-defined api key or an existing secret (`VLLM_API_KEY`).
+
+## Step 1: Preparing the Jaeger Configuration File
+
+1. Locate the example configuration files
+   `tutorials/assets/otel-example/{jaeger, jaeger-query, jaeger-collector}.yaml`.
+2. Open the files and examine the following fields:
+   - Specify your desired ports for `jaeger-collector` and `jaeger-query`
+   services for trace storage and retrieval, respectively.
+   - Verify that `COLLECTOR_OTLP_ENABLED` is set to `true`; this enables
+   Jaeger's native support for OpenTelemetry
+
+### Explanation of Key Items in Jaeger Files
+
+- **`ports`**: The ports through which Jaeger performs its operations
+(collection and querying in this case).
+- **`name`**: The unique identifier for your Jaeger deployment.
+- **`image`**: The single Docker image that runs all Jaeger's backend parts as
+well as its UI in a container.
+- **`type`**: The ClusterIP type ensures the query service is only reachable
+from within the cluster.
+
+## Step 2: Preparing the OpenTelemetry Collector File
+
+1. Locate the example collector files
+   `tutorials/assets/otel-example/{otel-collector, otel-collector-config}.yaml`.
+2. Open the file and examine the following fields:
+   - Specify the `receivers`, `processors`, and `exporters` of the collector
+   - The OpenTelemetry collection is an intermediary step of collecting traces
+   from the vLLM service and then providing these to Jaeger for its UI.
+   - The `resources` specify the compute resources available to the container
+   running the OpenTelemetry collector.
+
+## Step 3: Configuring Model and Monitoring
+
+1. Feel free to inspect the `values-12-otel-vllm.yaml` file. Note that the
+   `OTEL_EXPORTER_OTLP_ENDPOINT` specification enables metrics, traces, and
+   logs to be collected. Further configurations can be explored
+   [here](https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/).
+
+   Run the following from the `tutorials/assets` directory:
+
+   ```bash
+   sudo kubectl apply -f otel-example/jaeger.yaml
+   sudo kubectl apply -f otel-example/jaeger-collector.yaml
+   sudo kubectl apply -f otel-example/jaeger-query.yaml
+   sudo kubectl apply -f otel-example/otel-collector-config.yaml
+   sudo kubectl apply -f otel-example/otel-collector.yaml
+   sudo helm install vllm ../../helm/ -f values-12-otel-vllm.yaml
```

Comment:
@lucas-tucker : I don't see `values-12-otel-vllm.yaml`. Is it supposed to be there for this tutorial?